### ISSUE TITLE
stream_conv model lr get half in runtime

### ISSUE THREAD

**Issue by LeonKennedy** [facebookresearch/wav2letter#801]
### Question
i set my lr=0.4, but runtime log show only lr=0.2?

#### runtime log:
I0826 11:10:59.026504 13072 Train.cpp:151] Gflags after parsing
--flagfile=; --fromenv=; --tryfromenv=; --undefok=; --tab_completion_columns=80; --tab_completion_word=; --help=false; --helpfull=false; --helpmatch=; --helpon=; --helppackage=false; --helpshort=false; --helpxml=false; --version=false; --adambeta1=0.90000000000000002; --adambeta2=0.999; --am=; --am_decoder_tr_dropout=0; --am_decoder_tr_layerdrop=0; --am_decoder_tr_layers=1; --arch=am_qkids.arch; --archdir=/root/main/stream_conv; --attention=content; --attentionthreshold=2147483647; --attnWindow=no; --attnconvchannel=0; --attnconvkernel=0; --attndim=0; --batchsize=8; --beamsize=2500; --beamsizetoken=250000; --beamthreshold=25; --blobdata=false; --channels=1; --criterion=ctc; --critoptim=sgd; --datadir=/mnt/data/wav2letter/qkids-data/lists; --dataorder=input; --decoderattnround=1; --decoderdropout=0; --decoderrnnlayer=1; --decodertype=wrd; --devwin=0; --emission_dir=; --emission_queue_size=3000; --enable_distributed=false; --encoderdim=0; --eosscore=0; --eostoken=false; --everstoredb=false; --fftcachesize=1; --filterbanks=80; --flagsfile=/root/main/stream_conv/train_qkids.cfg; --framesizems=25; --framestridems=10; --gamma=1; --gumbeltemperature=1; --input=wav; --inputbinsize=100; --inputfeeding=false; --isbeamdump=false; --iter=9223372036854775807; --itersave=false; --labelsmooth=0; --leftWindowSize=50; --lexicon=/mnt/data/wav2letter/qkids-data/am/qkids-train+dev-unigram.lexicon; --linlr=-1; --linlrcrit=-1; --linseg=0; --lm=; --lm_memory=5000; --lm_vocab=; --lmtype=kenlm; --lmweight=0; --localnrmlleftctx=300; --localnrmlrightctx=0; --logadd=false; **--lr=0.40000000000000002**; --lr_decay=9223372036854775807; --lr_decay_step=9223372036854775807; --lrcosine=false; --lrcrit=0; --max_devices_per_node=8; --maxdecoderoutputlen=200; --maxgradnorm=0.5; --maxisz=33000; --maxload=-1; --maxrate=10; --maxsil=50; --maxtsz=9223372036854775807; --maxword=-1; --melfloor=1; --memstepsize=10485760; --mfcc=false; --mfcccoeffs=13; --mfsc=true; --minisz=200; --minrate=3; --minsil=0; --mintsz=2; --momentum=0; --netoptim=sgd; --noresample=false; --nthread=8; --nthread_decoder=1; --nthread_decoder_am_forward=1; --numattnhead=8; --onorm=target; --optimepsilon=1e-08; --optimrho=0.90000000000000002; --outputbinsize=5; --pctteacherforcing=100; --pcttraineval=1; --pow=false; --pretrainWindow=0; --replabel=0; --reportiters=1000; --rightWindowSize=50; --rndv_filepath=; --rundir=/mnt/data/wav2letter/qkids-data; --runname=sconv; --samplerate=16000; --sampletarget=0; --samplingstrategy=rand; --saug_fmaskf=27; --saug_fmaskn=2; --saug_start_update=-1; --saug_tmaskn=2; --saug_tmaskp=1; --saug_tmaskt=100; --sclite=; --seed=0; --show=false; --showletters=false; --silscore=0; --smearing=none; --smoothingtemperature=1; --softwoffset=10; --softwrate=5; --softwstd=5; --sqnorm=true; --stepsize=9223372036854775807; --surround=; --tag=; --target=tkn; --test=; --tokens=tokens.txt; --tokensdir=/mnt/data/wav2letter/qkids-data/am/; --train=train.lst; --trainWithWindow=false; --transdiag=0; --unkscore=-inf; --use_memcache=false; --uselexicon=true; --usewordpiece=true; --valid=dev.lst; --validbatchsize=-1; --warmup=1; --weightdecay=0; --wordscore=0; --wordseparator=_; --world_rank=0; --world_size=1; --alsologtoemail=; --alsologtostderr=false; --colorlogtostderr=false; --drop_log_memory=true; --log_backtrace_at=; --log_dir=; --log_link=; --log_prefix=true; --logbuflevel=0; --logbufsecs=30; --logemaillevel=999; --logfile_mode=436; --logmailer=/bin/mail; --logtostderr=true; --max_log_size=1800; --minloglevel=0; --stderrthreshold=2; --stop_logging_if_full_disk=false; --symbolize_stacktrace=true; --v=0; --vmodule=;
I0826 11:10:59.027776 13072 Train.cpp:152] Experiment path: /mnt/data/wav2letter/qkids-data/sconv
I0826 11:10:59.027787 13072 Train.cpp:153] Experiment runidx: 1
I0826 11:10:59.039278 13072 Train.cpp:199] Number of classes (network): 2609
I0826 11:10:59.146353 13072 Train.cpp:206] Number of words: 4207
I0826 11:10:59.153616 13072 Train.cpp:220] Loading architecture file from /root/main/stream_conv/am_qkids.arch
I0826 11:10:59.524392 13072 Train.cpp:252] [Network] Sequential [input -> (0) -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> (7) -> (8) -> (9) -> (10) -> (11) -> (12) -> (13) -> (14) -> (15) -> (16) -> (17) -> (18) -> (19) -> (20) -> (21) -> (22) -> (23) -> (24) -> (25) -> (26) -> (27) -> (28) -> (29) -> output]
        (0): View (-1 80 1 0)
        (1): SpecAugment ( W: 80, F: 27, mF: 2, T: 100, p: 1, mT: 2 )
        (2): Padding (0, { (5, 3), (0, 0), (0, 0), (0, 0), })
        (3): Conv2D (1->12, 10x1, 2,1, 0,0, 1, 1) (with bias)
        (4): ReLU
        (5): Dropout (0.100000)
        (6): LayerNorm ( axis : { 1 2 } , size : -1)
        (7): Time-Depth Separable Block (9, 80, 12) [960 -> 960 -> 960]
        (8): Time-Depth Separable Block (9, 80, 12) [960 -> 960 -> 960]
        (9): Padding (0, { (7, 1), (0, 0), (0, 0), (0, 0), })
        (10): Conv2D (12->15, 10x1, 2,1, 0,0, 1, 1) (with bias)
        (11): ReLU
        (12): Dropout (0.100000)
        (13): LayerNorm ( axis : { 1 2 } , size : -1)
        (14): Time-Depth Separable Block (11, 80, 15) [1200 -> 1200 -> 1200]
        (15): Time-Depth Separable Block (11, 80, 15) [1200 -> 1200 -> 1200]
        (16): Time-Depth Separable Block (11, 80, 15) [1200 -> 1200 -> 1200]
        (17): Padding (0, { (10, 0), (0, 0), (0, 0), (0, 0), })
        (18): Conv2D (15->21, 11x1, 1,1, 0,0, 1, 1) (with bias)
        (19): ReLU
        (20): Dropout (0.100000)
        (21): LayerNorm ( axis : { 1 2 } , size : -1)
        (22): Time-Depth Separable Block (11, 80, 21) [1680 -> 1680 -> 1680]
        (23): Time-Depth Separable Block (11, 80, 21) [1680 -> 1680 -> 1680]
        (24): Time-Depth Separable Block (11, 80, 21) [1680 -> 1680 -> 1680]
        (25): Time-Depth Separable Block (11, 80, 21) [1680 -> 1680 -> 1680]
        (26): Reorder (2,1,0,3)
        (27): View (1680 -1 1 0)
        (28): Linear (1680->2609) (with bias)
        (29): View (2609 0 -1 1)
I0826 11:10:59.524482 13072 Train.cpp:253] [Network Params: 39350858]
I0826 11:10:59.524518 13072 Train.cpp:254] [Criterion] ConnectionistTemporalClassificationCriterion
I0826 11:10:59.524554 13072 Train.cpp:262] [Network Optimizer] SGD
I0826 11:10:59.524559 13072 Train.cpp:263] [Criterion Optimizer] SGD
I0826 11:11:02.617501 13072 W2lListFilesDataset.cpp:141] 293764 files found.
I0826 11:11:02.634671 13072 Utils.cpp:102] Filtered 32420/293764 samples
I0826 11:11:02.846194 13072 W2lListFilesDataset.cpp:62] Total batches (i.e. iters): 32668
I0826 11:11:03.639250 13072 W2lListFilesDataset.cpp:141] 73726 files found.
I0826 11:11:03.645246 13072 Utils.cpp:102] Filtered 8123/73726 samples
I0826 11:11:03.686543 13072 W2lListFilesDataset.cpp:62] Total batches (i.e. iters): 8201
I0826 11:11:03.689311 13072 Train.cpp:568] Shuffling trainset
I0826 11:11:03.707760 13072 Train.cpp:575] Epoch 1 started!
I0826 11:52:40.505607 13072 Train.cpp:345] epoch:        1 | nupdates:         1000 | **lr: 0.200000** | lrcriterion: 0.000000 | runtime: 00:05:41 | bch(ms): 341.99 | smp(ms): 272.95 | fwd(ms): 23.03 | crit-fwd(ms): 3.39 | bwd(ms): 32.04 | optim(ms): 11.52 | loss:   17.97414 | train-TER: 93.44 | train-WER: 99.64 | dev.lst-loss:   16.19389 | dev.lst-TER: 95.41 | dev.lst-WER: 99.49 | avg-isz: 384 | avg-tsz: 010 | max-tsz: 043 | hrs:    8.53 | thrpt(sec/sec): 89.84
Memory Manager Stats
MemoryManager type: CachingMemoryManager
Number of allocated blocks:113
Size of free block pool (small):77
Size of free block pool (large):139
Total native mallocs:366
Total native frees:179

##### arch cfg

--runname=sconv
--rundir=/mnt/data/wav2letter/qkids-data
--datadir=/mnt/data/wav2letter/qkids-data/lists
--archdir=/root/main/stream_conv
--tokensdir=/mnt/data/wav2letter/qkids-data/am/
--train=train.lst
--valid=dev.lst
--lexicon=/mnt/data/wav2letter/qkids-data/am/qkids-train+dev-unigram.lexicon
--arch=am_qkids.arch
--tokens=tokens.txt
--criterion=ctc
--input=wav
--batchsize=8
--lr=0.4
--momentum=0
--reportiters=1000
--maxgradnorm=0.5
--nthread=8
--mfsc=true
--filterbanks=80
--usewordpiece=true
--wordseparator=_
--minisz=200
--mintsz=2
--maxisz=33000
--enable_distributed=false
--pcttraineval=1
--minloglevel=0
--logtostderr
--onorm=target
--sqnorm
--localnrmlleftctx=300

**Comment by tlikhomanenko** [comment]
Yep, this happens because of recent change in the way how we process lr_decay. Please set it's value say to  `--lr_decay=20000` (to avoid int overflow which happens right now). The fix will be in the flashlight where we moved w2l code.
